_pages/dev/ssr/server-side-rendering-optimization.md

```diff
@@ -24,22 +24,24 @@ Although these scenarios are fairly rare, they usually occur when the system is
 The SSR optimization engine addresses these issues as follows:
 
 - The optimization engine queues incoming requests
-- Only a certain number of queued pages are rendered before the rest of the queue defaults to Client-Side Rendering (CSR)
+- Only a certain number of queued pages are rendered before the rest of the queue defaults to Client-Side Rendering (CSR). With a new option available from Spartacus 3.4, the incoming requests can wait for the current render to finish instead of falling back to CSR.
 - Pages are served in SSR mode if they can be rendered in a given time (that is, within the time that is specified by a timeout)
 - If the engine falls back to CSR because the SSR render takes too long, once the SSR page is rendered, it is stored in memory and served with the subsequent request
-- The CSR app is served with the `Cache-Control:no-store` header to ensure it is not cached by the caching layer. 
+- The CSR app is served with the `Cache-Control:no-store` header to ensure it is not cached by the caching layer.
+- If the render is taking too long to finish, the engine will release its concurrency slot and notify about the hanging render. However, it will not release the resources taken by the hanging render. Therefore, these warnings should be taken seriously, as the server's resources could be quickly depleted if the route of the problem is not addressed in the application code.
 
-  **Note:** CSR renders should never be cached.
-- The rendered SSR pages should be cached (for example, using CDN) to ensure subsequent requests do not hit the SSR server. This reduces the server load and reduces CSR fallbacks to the least amount possible.
+  **Note:** CSR renders should _never_ be cached.
 
-***
+- The rendered SSR pages _should_ be cached (for example, using CDN) to ensure subsequent requests do not hit the SSR server. This reduces the server load and reduces CSR fallbacks to the least amount possible. To see the recommended SSR setup, please see [recommended SSR setup page](./server-side-rendering-setup.md).
```

Comment:
Would be good to note that using Spartacus SSR  'cache:true' option is not recommended for production.

Reply:
It is recommended to _not_ use it here: https://github.com/SAP/spartacus-docs/pull/1312/files#diff-3264b751e1ad68e616c03382c7d6c67ee5b7b55f5ff2e6b75fdc57f8dfaa4fe2R106

Let me know if you think we should emphasize it again here.